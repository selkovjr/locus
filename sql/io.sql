--
--  Locus datatype test
--
--
-- Testing the input and output functions
--

-- Whole contig
SELECT '1'::locus AS locus;
SELECT 'chr1'::locus AS locus;
SELECT 'GL383557.1'::locus AS locus;

-- Point position
SELECT '16:89831249'::locus AS locus;
SELECT ' 16 : 89831249 '::locus AS locus;
SELECT ' 16	:	89831249 '::locus AS locus;
SELECT '16:89,831,249'::locus AS locus;
SELECT ' 16 : 89,831,249 '::locus AS locus;
SELECT ' 16	:	89,831,249 '::locus AS locus;
SELECT 'chr16:89831249'::locus AS locus;
SELECT ' chr16 : 89831249 '::locus AS locus;
SELECT ' chr16	:	89831249 '::locus AS locus;
SELECT 'chr16:89,831,249'::locus AS locus;
SELECT ' chr16 : 89,831,249 '::locus AS locus;
SELECT ' chr16	:	89,831,249 '::locus AS locus;

-- Ranges
SELECT 'GL383557.1:100-500'::locus AS locus;
SELECT '16:89831249-89831439'::locus AS locus;
SELECT ' 16 : 89831249 - 89831439 '::locus AS locus;
SELECT ' 16 : 89831249	-	89831439 '::locus AS locus;
SELECT '16:89,831,249-89,831,439'::locus AS locus;
SELECT 'chr16:89831249-89831439'::locus AS locus;
SELECT 'chr16:89,831,249-89,831,439'::locus AS locus;

-- Open intervals
SELECT '16:1000000-'::locus AS locus;
SELECT 'chr16:1000000-'::locus AS locus;
SELECT 'chr16:1,000,000-'::locus AS locus;
SELECT '16:-89831249'::locus AS locus;
SELECT 'chr16:-89831249'::locus AS locus;
SELECT 'chr16:-89,831,249'::locus AS locus;

-- invalid input
SELECT 'ENST00000378344.2|ENSG00000158109.10'::locus AS locus;
SELECT 'ENST00000378344.2|ENSG00000158109.10:340'::locus AS locus;
SELECT 'chr16:89831249000000000000000000'::locus AS locus;
SELECT 'chr16:89831249000000'::locus AS locus;
SELECT '16:8983 1249-89831439'::locus AS locus;
SELECT '16:89831249-898 31439'::locus AS locus;
SELECT '16:89831249-89831,439'::locus AS locus;
SELECT 'chr16:garbage-89831249'::locus AS locus;
