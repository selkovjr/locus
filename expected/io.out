--
--  Locus datatype test
--
--
-- Testing the input and output functions
--
-- Whole contig
SELECT '1'::locus AS locus;
 locus
-------
 1
(1 row)

SELECT 'chr1'::locus AS locus;
 locus
-------
 chr1
(1 row)

SELECT 'GL383557.1'::locus AS locus;
   locus
------------
 GL383557.1
(1 row)

-- Point position
SELECT '16:89831249'::locus AS locus;
    locus
-------------
 16:89831249
(1 row)

SELECT ' 16 : 89831249 '::locus AS locus;
    locus
--------------
  16:89831249
(1 row)

SELECT ' 16	:	89831249 '::locus AS locus;
    locus
--------------
  16:89831249
(1 row)

SELECT '16:89,831,249'::locus AS locus;
    locus
-------------
 16:89831249
(1 row)

SELECT ' 16 : 89,831,249 '::locus AS locus;
    locus
--------------
  16:89831249
(1 row)

SELECT ' 16	:	89,831,249 '::locus AS locus;
    locus
--------------
  16:89831249
(1 row)

SELECT 'chr16:89831249'::locus AS locus;
     locus
----------------
 chr16:89831249
(1 row)

SELECT ' chr16 : 89831249 '::locus AS locus;
      locus
-----------------
  chr16:89831249
(1 row)

SELECT ' chr16	:	89831249 '::locus AS locus;
      locus
-----------------
  chr16:89831249
(1 row)

SELECT 'chr16:89,831,249'::locus AS locus;
     locus
----------------
 chr16:89831249
(1 row)

SELECT ' chr16 : 89,831,249 '::locus AS locus;
      locus
-----------------
  chr16:89831249
(1 row)

SELECT ' chr16	:	89,831,249 '::locus AS locus;
      locus
-----------------
  chr16:89831249
(1 row)

-- Ranges
SELECT 'GL383557.1:100-500'::locus AS locus;
       locus
--------------------
 GL383557.1:100-500
(1 row)

SELECT '16:89831249-89831439'::locus AS locus;
        locus
----------------------
 16:89831249-89831439
(1 row)

SELECT ' 16 : 89831249 - 89831439 '::locus AS locus;
         locus
-----------------------
  16:89831249-89831439
(1 row)

SELECT ' 16 : 89831249	-	89831439 '::locus AS locus;
         locus
-----------------------
  16:89831249-89831439
(1 row)

SELECT '16:89,831,249-89,831,439'::locus AS locus;
        locus
----------------------
 16:89831249-89831439
(1 row)

SELECT 'chr16:89831249-89831439'::locus AS locus;
          locus
-------------------------
 chr16:89831249-89831439
(1 row)

SELECT 'chr16:89,831,249-89,831,439'::locus AS locus;
          locus
-------------------------
 chr16:89831249-89831439
(1 row)

-- Open intervals
SELECT '16:1000000-'::locus AS locus;
    locus
-------------
 16:1000000-
(1 row)

SELECT 'chr16:1000000-'::locus AS locus;
     locus
----------------
 chr16:1000000-
(1 row)

SELECT 'chr16:1,000,000-'::locus AS locus;
     locus
----------------
 chr16:1000000-
(1 row)

SELECT '16:-89831249'::locus AS locus;
    locus
--------------
 16:-89831249
(1 row)

SELECT 'chr16:-89831249'::locus AS locus;
      locus
-----------------
 chr16:-89831249
(1 row)

SELECT 'chr16:-89,831,249'::locus AS locus;
      locus
-----------------
 chr16:-89831249
(1 row)

-- invalid input
SELECT 'ENST00000378344.2|ENSG00000158109.10'::locus AS locus;
ERROR:  Conting name can't be longer than 15 characters
LINE 1: SELECT 'ENST00000378344.2|ENSG00000158109.10'::locus AS locu...
               ^
SELECT 'ENST00000378344.2|ENSG00000158109.10:340'::locus AS locus;
ERROR:  Conting name can't be longer than 15 characters
LINE 1: SELECT 'ENST00000378344.2|ENSG00000158109.10:340'::locus AS ...
               ^
SELECT 'chr16:89831249000000000000000000'::locus AS locus;
ERROR:  invalid number 89831249000000000000000000 (long overflow)
LINE 1: SELECT 'chr16:89831249000000000000000000'::locus AS locus;
               ^
SELECT 'chr16:89831249000000'::locus AS locus;
ERROR:  invalid number 89831249000000 (int32 overflow)
LINE 1: SELECT 'chr16:89831249000000'::locus AS locus;
               ^
SELECT '16:8983 1249-89831439'::locus AS locus;
ERROR:  bad locus representation
LINE 1: SELECT '16:8983 1249-89831439'::locus AS locus;
               ^
DETAIL:  syntax error at or near "1249"
SELECT '16:89831249-898 31439'::locus AS locus;
ERROR:  bad locus representation
LINE 1: SELECT '16:89831249-898 31439'::locus AS locus;
               ^
DETAIL:  syntax error at or near "31439"
SELECT '16:89831249-89831,439'::locus AS locus;
ERROR:  locus syntax error
LINE 1: SELECT '16:89831249-89831,439'::locus AS locus;
               ^
DETAIL:   bad character ,
SELECT 'chr16:garbage-89831249'::locus AS locus;
ERROR:  locus syntax error
LINE 1: SELECT 'chr16:garbage-89831249'::locus AS locus;
               ^
DETAIL:   bad character g
